executor/api/grpc/kfserving/server.go

```diff
@@ -0,0 +1,128 @@
+	reqPayload := payload.ProtoPayload{Msg: request}
+	resPayload, err := seldonPredictorProcess.Status(g.predictor.Graph, request.Name, &reqPayload)
+	if err != nil {
+		return nil, err
+	}
+	return resPayload.GetPayload().(*proto.ModelReadyResponse), nil
+}
+
+func (g GrpcKFServingServer) ServerMetadata(ctx context.Context, request *proto.ServerMetadataRequest) (*proto.ServerMetadataResponse, error) {
+	panic("implement me")
+}
+
+func (g GrpcKFServingServer) ModelMetadata(ctx context.Context, request *proto.ModelMetadataRequest) (*proto.ModelMetadataResponse, error) {
+	md := grpc.CollectMetadata(ctx)
+	ctx = context.WithValue(ctx, payload.SeldonPUIDHeader, md.Get(payload.SeldonPUIDHeader)[0])
+	seldonPredictorProcess := predictor.NewPredictorProcess(ctx, g.Client, logf.Log.WithName("infer"), g.ServerUrl, g.Namespace, md)
+	reqPayload := payload.ProtoPayload{Msg: request}
+	resPayload, err := seldonPredictorProcess.Metadata(g.predictor.Graph, request.Name, &reqPayload)
+	if err != nil {
+		return nil, err
+	}
+	return resPayload.GetPayload().(*proto.ModelMetadataResponse), nil
+}
+
+func (g GrpcKFServingServer) ModelInfer(ctx context.Context, request *proto.ModelInferRequest) (*proto.ModelInferResponse, error) {
+	md := grpc.CollectMetadata(ctx)
+	ctx = context.WithValue(ctx, payload.SeldonPUIDHeader, md.Get(payload.SeldonPUIDHeader)[0])
+	seldonPredictorProcess := predictor.NewPredictorProcess(ctx, g.Client, logf.Log.WithName("infer"), g.ServerUrl, g.Namespace, md)
+	reqPayload := payload.ProtoPayload{Msg: request}
+	resPayload, err := seldonPredictorProcess.Predict(g.predictor.Graph, &reqPayload)
+	if err != nil {
+		return nil, err
+	}
+	return resPayload.GetPayload().(*proto.ModelInferResponse), nil
+}
+
+func (g GrpcKFServingServer) ModelStreamInfer(server proto.GRPCInferenceService_ModelStreamInferServer) error {
+	panic("implement me")
+}
+
+func (g GrpcKFServingServer) ModelConfig(ctx context.Context, request *proto.ModelConfigRequest) (*proto.ModelConfigResponse, error) {
+	panic("implement me")
+}
+
+func (g GrpcKFServingServer) ModelStatistics(ctx context.Context, request *proto.ModelStatisticsRequest) (*proto.ModelStatisticsResponse, error) {
+	panic("implement me")
+}
+
+func (g GrpcKFServingServer) RepositoryIndex(ctx context.Context, request *proto.RepositoryIndexRequest) (*proto.RepositoryIndexResponse, error) {
+	panic("implement me")
+}
+
+func (g GrpcKFServingServer) RepositoryModelLoad(ctx context.Context, request *proto.RepositoryModelLoadRequest) (*proto.RepositoryModelLoadResponse, error) {
+	panic("implement me")
+}
+
+func (g GrpcKFServingServer) RepositoryModelUnload(ctx context.Context, request *proto.RepositoryModelUnloadRequest) (*proto.RepositoryModelUnloadResponse, error) {
+	panic("implement me")
+}
+
+func (g GrpcKFServingServer) SystemSharedMemoryStatus(ctx context.Context, request *proto.SystemSharedMemoryStatusRequest) (*proto.SystemSharedMemoryStatusResponse, error) {
```

Comment:
I guess these are all the Triton-specific endpoints that are brought by extensions. Some of them seem very specific to CUDA. 

What are your thoughts on these @cliveseldon ? Should we implement them? I guess we could always tell users to run in `noEngine` mode if they want to leverage these?

Reply:
yes I suppose as they are not part of our graph protocol